projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/monitoring/DataJobMonitor.java

```diff
@@ -198,8 +198,12 @@ void recordJobExecutionStatus(KubernetesService.JobExecution jobStatus) {
             updateDataJobTerminationStatusGauge(dataJob);
         }
 
-        var updatedExecution = jobExecutionService.updateJobExecution(dataJob, jobStatus, executionResult);
-        updatedExecution.ifPresent(jobsService::updateLastExecution);
+        // Update the job execution
+        jobExecutionService.updateJobExecution(dataJob, jobStatus, executionResult);
```

Comment:
if these two operations are not in a transaction we might have inconsistent execution state.

Reply:
Wasn't that the purpose of the `@Transactional` annotation?